packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -184,6 +194,27 @@ export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
       })
       .next(() => changedDocs);
   }
+
+  /**
+   * Removes all changes in the remote document changelog through `changeId`
+   * (inclusive).
+   */
+  removeDocumentChangesThroughChangeId(
+    transaction: PersistenceTransaction,
+    changeId: number
+  ): PersistencePromise<void> {
+    assert(
+      changeId <= this._lastProcessedDocumentChangesId,
+      'Cannot remove unprocessed changes'
+    );
+
+    if (changeId === 0) {
```

Comment:
Is this an optimization?  Is it warranted (are we going to hit this frequently or in critical paths)?

Reply:
This gets hit during startup, but only for the very first time. It's probably ok to remove this.